### PR TITLE
Purifies Hparams class by removing some fields

### PR DESCRIPTION
**Pull request by milocress**
removed `datadir` and `is_test` fields from hparams and put them into individual classes

### CODE REVIEW

File: composer/datasets/dataset_hparams.py [hunk 1]
```diff
@@ -120,15 +120,12 @@ class DatasetHparams(hp.Hparams, abc.ABC):
...
-    is_train: bool = hp.optional('Whether to load the training data (the default) or validation data.', default=True)
     drop_last: bool = hp.optional(textwrap.dedent("""\
         If the number of samples is not divisible by the batch size,
         whether to drop the last batch (the default) or pad the last batch with zeros."""),
                                   default=True)
     shuffle: bool = hp.optional('Whether to shuffle the dataset for each epoch. Defaults to True.', default=True)
 
-    datadir: Optional[str] = hp.optional('The path to the data directory', default=None)
-
     @abc.abstractmethod
```
Comment (ravi-mosaicml) [inline composer/datasets/dataset_hparams.py:129]: Cam upi also remove the "datadir" and "is_train" arguments from the docstring above?

**Review by ravi-mosaicml — APPROVED** [review]
LGTM! See comment.

File: composer/datasets/dataset_hparams.py (hunk 1 above)
Comment (milocress) [reply composer/datasets/dataset_hparams.py:129]: yes, good catch

File: composer/datasets/c4_hparams.py [hunk 2]
```diff
@@ -49,6 +49,8 @@ class StreamingC4Hparams(DatasetHparams):
     mlm: bool = hp.optional('Whether or not to use masked language modeling.', default=False)
     mlm_probability: float = hp.optional('If `mlm==True`, the probability that tokens are masked.', default=0.15)
 
+    datadir: Optional[str] = hp.optional('The path to the data directory', default=None)
```
Comment (ravi-mosaicml) [inline composer/datasets/c4_hparams.py:52]: I don't think this class uses the `datadir`.

**Review by ravi-mosaicml — COMMENTED** [review]
Whoops -- forgot about the docstrings. Could you make sure that `datadir` and `is_train` have docstring entries for any classes where you added in these args?

File: composer/datasets/ade20k_hparams.py [hunk 3]
```diff
@@ -48,6 +49,9 @@ class ADE20kDatasetHparams(DatasetHparams, SyntheticHparamsMixin):
     final_size: int = hp.optional('Final size of the image and target', default=512)
     ignore_background: bool = hp.optional('If true, ignore the background class in training loss', default=True)
 
+    datadir: Optional[str] = hp.optional('The path to the data directory', default=None)
+    is_train: bool = hp.optional('Whether to load the training data (the default) or validation data.', default=True)
```
Comment (ravi-mosaicml) [inline composer/datasets/ade20k_hparams.py:53]: I don't think this class needs `is_train`

Comment (milocress) [reply composer/datasets/ade20k_hparams.py:53]: the constructor is invoked with `is_train=False` in `test_dataset_registry.py`

Comment (ravi-mosaicml) [reply composer/datasets/ade20k_hparams.py:53]: Can we fix that in the tests? Would prefer not to keep non-test codes around to support tests.

Comment (milocress) [reply composer/datasets/ade20k_hparams.py:53]: yes

**Comment by milocress** [comment]
removed `datadir` and `is_test` fields from `hparams` and put them into individual classes

**Pull request merged**